File: backend/auth/src/src/auth/dtos/user.dto.ts
import { User } from '../entities/user.entity';

export class UserDto {
  id: string;
  email: string;
  username: string;

  public static mapUser(user: User): UserDto {
    const { id, email, username } = user;
    return { id, email, username };
  }
}

File: backend/java/api-stream/src/src/main/java/org/deus/src/dtos/fromModels/UserDTO.java
package org.deus.src.dtos.fromModels;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {
    private String id;
    private String email;
    private String username;
}
